### PR TITLE
HW35 Aliaksandr

### PR DESCRIPTION
**Pull request by AlexShcherba26**

### CODE REVIEW

File: hw-35.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        hasChildren: true,
+        hasEducation: false
+    }
+]
+
+// 1. Создать строку из имен пользователей через запятую
+
+console.log("1. Создать строку из имен пользователей через запятую")
+
+const getUsers = (arr: User[]) => {
```
Comment (jathpr) [inline hw-35.ts:43]: prefer explicit names as users against common arr

File: hw-35.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+console.log(getUsers(users))
+
+// 2. Посчитать общее количнство машин у пользователей
+
+console.log("\n2. Посчитать общее количнство машин у пользователей")
+
+const totalCars = (arr: User[]): number => {
+    return arr.map(user => user.cars).length
```
Comment (jathpr) [inline hw-35.ts:56]: wrong, it will be always the same size as users. And do not depend on cars count

File: hw-35.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+console.log(totalCars(users))
+
+// 3. Создать функцию, которая бы принимала массив пользователей и отфильтровывала пользователей на наличие образования
+
+console.log("\n3. Создать функцию, которая бы принимала массив пользователей и отфильтровывала пользователей на наличие образования")
+
+const isEducation = (arr: User[]) => {
```
Comment (jathpr) [inline hw-35.ts:65]: function with names starting from `is` usually return boolean value

Comment (jathpr) [reply hw-35.ts:65]: and you don't use arr parameter here

File: hw-35.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    const userList = [...users];
```
Comment (jathpr) [inline hw-35.ts:66]: filter doesn't mutate original array, so you don't need to copy it before filter

File: hw-35.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+console.log(isEducation(users))
+
+// 4. Создать функцию, которая бы принимала массив пользователей и отфильтровывала пользователей на наличие животных
+
+console.log("\n4. Создать функцию, которая бы принимала массив пользователей и отфильтровывала пользователей на наличие животных")
+
+const hasAnimals = (arr: User[]) => {
+    const userList = [...users];
```
Comment (jathpr) [inline hw-35.ts:78]: same

File: hw-35.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+console.log("\n5. Создать функцию, которая бы принимала массив пользователей и отдавала бы  строку с названиями марок автомобилей через запятую")
+
+const carName = (arr: User[]) => {
+    const carBrands: string[] = [];
+
+    for (const user of arr) {
+        if (user.cars) {
+            for (const car of user.cars) {
+                if (!carBrands.includes(car)) {
+                    carBrands.push(car);
```
Comment (jathpr) [inline hw-35.ts:96]: nice!

**Review by jathpr — CHANGES_REQUESTED** [review]
good in general

File: hw-35.ts (hunk 1 above)
Comment (AlexShcherba26) [reply hw-35.ts:43]: fixed

File: hw-35.ts (hunk 4 above)
Comment (AlexShcherba26) [reply hw-35.ts:66]: fixed

File: hw-35.ts (hunk 5 above)
Comment (AlexShcherba26) [reply hw-35.ts:78]: fixed

File: hw-35.ts (hunk 2 above)
Comment (AlexShcherba26) [reply hw-35.ts:56]: changed the code

File: hw-35.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+console.log(getUsers(users))
+
+// 2. Посчитать общее количнство машин у пользователей
+
+console.log("\n2. Посчитать общее количнство машин у пользователей")
+
+const totalCars = (users: User[]) => {
+    return users.reduce((acc, user) => {
+        if (user.cars) {
```
Comment (jathpr) [inline hw-35.ts:57]: common way with the same result 
`=> user.cars ? acc + user.cars.length : acc`

**Review by jathpr — APPROVED** [review]
E!

**Pull request merged**